### PR TITLE
[WIP] Deployer

### PR DESCRIPTION
**Pull request by justaugustus**
Very rough cut. Do not merge.

### CODE REVIEW

File: pkg/apis/azureprovider/v1alpha1/register.go [hunk 1]
```diff
@@ -18,14 +19,18 @@ limitations under the License.
 // Package v1alpha1 contains API Schema definitions for the azureprovider v1alpha1 API group
 // +k8s:openapi-gen=true
 // +k8s:deepcopy-gen=package,register
-// +k8s:conversion-gen=github.com/platform9/azure-provider/pkg/apis/azureprovider
+// +k8s:conversion-gen=sigs.k8s.io/cluster-api-provider-aws/pkg/apis/azureprovider
```
Comment (soggiest) [inline pkg/apis/azureprovider/v1alpha1/register.go]: We should change this to azure

Comment (justaugustus) [reply pkg/apis/azureprovider/v1alpha1/register.go]: Will do

File: azure-pipelines.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ variables:
 - name: GOPATH # Go workspace path
   value: '$(system.defaultWorkingDirectory)/gopath'
 - name: modulePath # Path to the module's code
-  value: '$(GOPATH)/src/github.com/platform9/azure-provider'
+  value: '$(GOPATH)/src/github.com/platform9/cluster-api-provider-azure'
 - name: KubernetesVersion
   value: 'v1.12.2'
```
Comment (tariq1890) [inline azure-pipelines.yml]: This needs changed as well I assume?
```suggestion
   value: 'v1.13.1'
```

**Comment by tariq1890** [comment]
Looks good so far. Additionally could you run a `dep ensure -update` since you have used the latest versions of Azure Go SDK and go-autorest. This may be required to trigger a run  of `go get -u github.com/golang/{protoc,gen}` in the vendor dirs. If this has already been done and the builds work fine, please disregard :).

**Comment by justaugustus** [comment]
Yep. I did it a little earlier in the process, but I'll do one more just to be safe! :)

**Comment by tariq1890** [comment]
@justaugustus We would also need to change the `azure-pipelines.yml` too I guess?

**Comment by justaugustus** [comment]
@tariq1890 -- with the move into the kubernetes-sigs org, we'll need to reassess the testing framework, so I haven't touched anything related to Azure pipelines. Once this is in a state that builds, I can double back and take a look at it.

**Comment by tariq1890** [comment]
@justaugustus Makes sense. We'd definitely want to use the k8s test-infra. I was hoping we could use some CI check to verify this build. I guess we can make do with the manual verification. I can pull your branch and validate.

**Comment by juan-lee** [comment]
@justaugustus is PR still WIP? What else needs to get done for this to merge?

**Comment by justaugustus** [comment]
@tariq1890 @juan-lee -- still a work in progress at the moment. I'm hoping to have an update in early next week, if not sooner.

File: Gopkg.toml [hunk 3]
```diff
@@ -28,28 +48,6 @@ required = [
   name = "github.com/Azure/go-autorest"
   version = "11.2.8"
 
-[[constraint]]
-  name = "github.com/ghodss/yaml"
-  version = "1.0.0"
-
-[[constraint]]
```
Comment (awesomenix) [inline Gopkg.toml]: We might want to retain a specific version of cluster-api

**Comment by justaugustus** [comment]
@tariq1890 @juan-lee @soggiest -- I'm declaring bankruptcy on this PR. Holiday me was a little too ambitious in reimplementing this in a few days.

I also don't want to set the bad example of introducing PRs with large, sweeping changes across the codebase. I'll be working to chunk down the work on this branch into smaller atomic units and will issue separate PRs for them.

/close

**Pull request closed without merging**